src/codegen/encoding/scale_encoding.rs

```diff
@@ -578,4 +580,60 @@ impl AbiEncoding for ScaleEncoding {
     fn is_packed(&self) -> bool {
         self.packed_encoder
     }
+
+    /// TODO: This is used and tested for error data (Error and Panic) only.
+    fn const_encode(&self, args: &[Expression]) -> Option<Vec<u8>> {
+        let mut result = vec![];
+        for arg in args {
+            match arg {
+                Expression::AllocDynamicBytes {
+                    initializer: Some(data),
+                    ty: Type::String,
+                    ..
+                } => result.extend_from_slice(&String::from_utf8(data.to_vec()).unwrap().encode()),
```

Comment:
This could be simply `result.extend_from_slice(&data.encode()),`

Reply:
`data` is a vector, not a string. Just because it coincidentally works I still think it's better to encode it using the correct type

Reply:
You're assuming that the contents is valid utf8, which it does not need to be.

Reply:
Hmm interesting, do we allow UTF16? Or why is that?

Reply:
I changed it to use them same code for both types for now

Reply:
```solidity
revert("abc \xfe def");
```
is not valid unicode but it is a string